service/src/main/java/org/mskcc/smile/service/impl/RequestServiceImpl.java

```diff
@@ -78,6 +78,19 @@ public Boolean saveRequest(SmileRequest request) throws Exception {
             }
             requestRepository.save(request);
             return Boolean.TRUE;
+        } else if ((requestHasMetadataUpdates(savedRequest.getLatestRequestMetadata(),
+                request.getLatestRequestMetadata()))
+                || savedRequest.getSmileSampleList().size() != request.getSmileSampleList().size()) {
```

Comment:
I'm hesitant to depend on comparing the sample sizes here in case they _do match_ even if they aren't the same samples.. do we need this check actually? if we're adding the sample-to-request relationship in the sample service?